
Don't output "add %gs:0x0,%eax" as part of a TLS access

This instruction is currently not accepted by the validator,
although the validator is being overly cautious here.

BUG: http://code.google.com/p/nativeclient/issues/detail?id=263

diff --git a/gcc/config/i386/i386.md b/gcc/config/i386/i386.md
index d9feab7..cfb4b16 100644
--- gcc-4.4.3/gcc/config/i386/i386.md
+++ gcc-4.4.3/gcc/config/i386/i386.md
@@ -14920,7 +14920,7 @@
 	(plus:SI (unspec:SI [(const_int 0)] UNSPEC_TP)
 		 (match_operand:SI 1 "register_operand" "0")))
    (clobber (reg:CC FLAGS_REG))]
-  "!TARGET_64BIT"
+  "!TARGET_64BIT && !flag_control_integrity"
   "add{l}\t{%%gs:0, %0|%0, DWORD PTR %%gs:0}"
   [(set_attr "type" "alu")
    (set_attr "modrm" "0")
